### DISCUSSION TITLE
Can REM support pool cover control?

### DISCUSSION THREAD

**Discussion by dkossman** [tagyoureit/nodejs-poolController#975] [REM]
Hello,

apologies if this is answered somewhere already but I have a couple of relays on my I2C relay board that control my pool cover in parallel with the manual key switch.  In my existing pool server code, clicking the "open" control on my web interface (or telling Siri to open the pool cover, via Homebridge)) turns the relay on and starts a timer that turns it off after about a minute, so simulating a person holding the key switch til the cover is fully open.  ditto for the "close" control.

I can see in REM that there are programmable off delays for the relays, but the maximum delay seems to be 5,000 msec, and I need between 44,000 and 77,000.

Even if REM was updated to allow longer delays, would this allow me to control the cover relays as noted above?  (i.e. click a control on the dashPanel UI and have the relay activate, then deactivate after N seconds). 

FYI other relays on this board control the spa valves and the spa bubbler. 

thanks in advance for any insights.

Don

**Comment by rstrouse** [comment]
Typically, with pool covers there is an end stop with a dry contact.  When the roller reaches the end at either side it should trigger open or closed.  This ensures certain features will not engage when the cover is closed.

**Comment by dkossman** [comment]
yes, the pool cover operating circuit has 2 dry contacts (magnetic reed switches) that stop the motor when the cover reaches one end or the other.

the issue is that the open and close relays must be released once the cover is opened or closed- they cannot  be held closed.  otherwise the manual keyswitch that also operates the cover will not work, which is a major problem.  it is also a potential safety issue, as if something goes wrong with one of the dry contacts, the cover could close with swimmers in the pool.

so the question remains whether REM has a mechanism that can activate a relay, and then deactivate it after a delay of N seconds, where N can be as long as 77 seconds?

thanks as always for your help!!

regards

**Reply by rstrouse** [reply]
Hmmm.  Time is really not the answer with this.  Typically pool covers are not automated like this and are only sensor driven meaning a detection that it is open or closed via those contacts.  I don't know if this is because of all the safety concerns about moving the cover when not standing next to the motor or if the correct way to control it is by momentary switch.  Button held down in the direction of movement.

**Reply by dkossman** [reply]
You make a good point - is there a way to implement “activate relay while button is held down”? 

Thanks
